### PR TITLE
ignore NAs when pseudocount is TRUE

### PR DESCRIPTION
**Pull request by 0xMuluh**
Ignore NAs and perform calculation automatically.

ping: #489

### CODE REVIEW

File: R/transformCounts.R [hunk 1]
```diff
@@ -361,15 +362,20 @@ setMethod("transformAssay", signature = c(x = "SummarizedExperiment"),
...
+        # If pseudocount TRUE but some negative values, numerical pseudocount needed
+        if ( pseudocount && any(mat < 0, na.rm = TRUE) ){
+            stop("The assay contains negative values. ",
                  "'pseudocount' must be specified manually.", call. = FALSE)
         }
         # If pseudocount TRUE, set it to  half of non-zero minimum value
         # else set it to zero.
         # Get min value
-        value <- min(mat[mat>0])
+        value <- min(mat[mat > 0], na.rm = TRUE)
```
Comment (antagomir) [inline R/transformCounts.R]: I think we could ultimately leave the decision to user so na.rm could be a parameter that is passed with ...

Comment (antagomir) [reply R/transformCounts.R]: but it could have default TRUE in the user function

**Review by antagomir — APPROVED** [review]
Seems good to me.

**Comment by antagomir** [comment]
The unit test should test the actual calculation and compare it to cases that are calculated manually, checking that the real use scenarios work properly. Can you add that?

File: R/transformCounts.R [hunk 2]
```diff
@@ -13,7 +13,8 @@
 #'   (Default: \code{"samples"})
 #' 
 #' @param pseudocount \code{Logical scalar} or \code{numeric scalar}. When TRUE,
-#'   automatically adds half of the minimum positive value of \code{assay.type}.
+#'   missing values(NAs) are ignored and, it automatically adds half of the 
+#'   minimum positive value of \code{assay.type}.
```
Comment (TuomasBorman) [inline R/transformCounts.R]: Missing values is just a side note. Better description might be

When \code{TRUE}, automatically adds half of the minimum positive value of \code{assay.type} (missing values ignored by default: \code{na.rm = TRUE}).

**Review by TuomasBorman — CHANGES_REQUESTED** [review]
vegan::decostand has also na.rm parameter. Check that this is not causing problems. (I think this not causing problems even though same value handles both pseudocount and vegan::decostand)

File: R/transformCounts.R (hunk 2 above)
Comment (0xMuluh) [reply R/transformCounts.R]: Thanks

File: R/transformCounts.R [hunk 3]
```diff
@@ -359,17 +361,22 @@ setMethod("transformAssay", signature = c(x = "SummarizedExperiment"),
...
-        if ( pseudocount && (any(is.na(mat)) || any(mat < 0, na.rm = TRUE)) ){
-            stop("The assay contains missing or negative values. ",
+        # If pseudocount TRUE and some NAs, a warning is issued
+        if ( pseudocount && any(is.na(mat)) ){
+            warning("The assay contains missing values (NAs). These will be 
+                ignored in pseudocount calculation.", call. = FALSE)
+        }    
+        # If pseudocount TRUE but some negative values, numerical pseudocount needed
+        if ( pseudocount && any(mat < 0, na.rm = TRUE) ){
+            stop("The assay contains missing values (NAs) or negative values. ",
```
Comment (TuomasBorman) [inline R/transformCounts.R]: This message is now incorrect. It should be just "The assay contains negative values.."

After this fix, good to go

**Review by TuomasBorman — APPROVED** [review]

**Pull request merged**